### ISSUE TITLE
Fatal error when trying to edit row with row checked and button under the table

### ISSUE THREAD

**Issue by raimundmeyer** [phpmyadmin/phpmyadmin#15571]
### Describe the bug

When selecting one or several rows with the checkbox in the row view and clicking the edit button, a fatal PHP  error is thrown

### To Reproduce

1. Go to browse view of a table that has entries
2. check  the checkbox in the row
3. click "Edit" below the table view
4. PHP error: 

>  PHP Fatal error:  Uncaught TypeError: Argument 3 passed to PhpMyAdmin\InsertEdit::urlParamsInEditMode() must be of the type string or null, array given, called in /var/www/phpmyadmin/libraries/entry_points/table/change.php on line 142 and defined in /var/www/phpmyadmin/libraries/classes/InsertEdit.php:254
Stack trace:
#0 /var/www/phpmyadmin/libraries/entry_points/table/change.php(142): PhpMyAdmin\InsertEdit->urlParamsInEditMode(Array, Array, Array)
#1 /var/www/phpmyadmin/libraries/entry_points/table/row_action.php(97): include('/var/www/phpmya...')
#2 /var/www/phpmyadmin/libraries/routes.php(511): require_once('/var/www/phpmya...')
#3 /var/www/phpmyadmin/index.php(58): {closure}(Array)
#4 {main}
  thrown in /var/www/phpmyadmin/libraries/classes/InsertEdit.php on line 254

### Expected behavior

It should open the edit view

### Server configuration
Latest git: Git revision: b9380c5 from master branch,
committed on Nov 13, 2019 at 10:05 AM by Hosted Weblate,
authored on Nov 10, 2019 at 11:41 PM by Allan Nordhøy

https://demo.phpmyadmin.net/master/index.php

**Comment by Tithugues** [comment]
As the 3rd parameter is not used in the function, I think we can just remove it. We won't have the error anymore.

**Comment by AshwinGinoria** [comment]
Hello, I would like to help but this would be my first contribution so can somebody guide me a bit ??

**Comment by williamdes** [comment]
Follow the steps:
- Fork the repository
- Clone the fork locally into a web folder eg:`/var/www`
- Checkout QA_4_9 branch since it is an bug fix and create a new branch from QA_4_9 
- Run `composer install`
- Create a `config.inc.php` if needed using the example file 
- Access your phpmyadmin website and login
- Start developing the fix
- Do not forget to use [`--signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) option to add your signoff line in the commit body, please use your real name (this option is an agreement to our [policy](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md#patches-submission))
- Push your changes and open a pull-request to this repository, branch: QA_4_9 

Do you have any questions?

**Comment by AshwinGinoria** [comment]
The QA_4_9 branch of My fork is showing "This branch is 1 commit behind". Is that a problem ?
If yes how to remove it ??

**Comment by williamdes** [comment]
>  Is that a problem ?

I would say no in most cases

You can do:
```bash
git remote add phpmyadmin https://github.com/phpmyadmin/phpmyadmin.git
git fetch --all
git checkout QA_4_9
git reset --hard QA_4_9 phpmyadmin/QA_4_9
git push --force QA_4_9
```
Warning: I hope you followed the advice to create a branch for the fix because the script will reset and sync the `QA_4_9` branch as it is on our repo.

**Comment by AshwinGinoria** [comment]
I was following the installation instructions given in the documentation.
the repository got cloned successfully but I am getting the following error when running composer update. I tried googling it but couldn't find much there. please Help 
![Screenshot from 2019-11-17 15-42-25](https://user-images.githubusercontent.com/43852150/69006164-4f8f1700-0951-11ea-9326-bd1ae117836b.png)

**Comment by williamdes** [comment]
@AshwinGinoria you seem to have a connexion issue, try another time 
We can not help much but re-trying is the best option, if you have a bad Internet connexion it can be difficult to pass this step without some patience :)

**Comment by AshwinGinoria** [comment]
Done!! :D 
I was on my college network which uses a proxy server so I switched to my phone's hotspot.

**Comment by AshwinGinoria** [comment]
phpmyadmin is now installed but it is unable to connect to mysql. The error I'm facing is 

![Screenshot from 2019-11-18 03-43-59](https://user-images.githubusercontent.com/43852150/69015039-22288480-09b6-11ea-9e22-38faabe609f0.png)

This is my config.inc.php file

`<?php
$cfg['blowfish_secret'] = 'xampp';
$i = 0;
$i++;

/* Authentication type and info */
$cfg['Servers'][$i]['auth_type'] = 'config';
$cfg['Servers'][$i]['user'] = 'root';
$cfg['Servers'][$i]['password'] = '';
$cfg['Servers'][$i]['extension'] = 'mysqli';
$cfg['Servers'][$i]['AllowNoPassword'] = true;
$cfg['Lang'] = '';

/* Bind to the localhost ipv4 address and tcp */
$cfg['Servers'][$i]['host'] = '127.0.0.1';
$cfg['Servers'][$i]['connect_type'] = 'tcp';

/* User for advanced features */
$cfg['Servers'][$i]['controluser'] = 'pma';
$cfg['Servers'][$i]['controlpass'] = '';

/* Advanced phpMyAdmin features */
$cfg['Servers'][$i]['pmadb'] = 'phpmyadmin';
$cfg['Servers'][$i]['bookmarktable'] = 'pma__bookmark';
$cfg['Servers'][$i]['relation'] = 'pma__relation';
$cfg['Servers'][$i]['table_info'] = 'pma__table_info';
$cfg['Servers'][$i]['table_coords'] = 'pma__table_coords';
$cfg['Servers'][$i]['pdf_pages'] = 'pma__pdf_pages';
$cfg['Servers'][$i]['column_info'] = 'pma__column_info';
$cfg['Servers'][$i]['history'] = 'pma__history';
$cfg['Servers'][$i]['designer_coords'] = 'pma__designer_coords';
$cfg['Servers'][$i]['tracking'] = 'pma__tracking';
$cfg['Servers'][$i]['userconfig'] = 'pma__userconfig';
$cfg['Servers'][$i]['recent'] = 'pma__recent';
$cfg['Servers'][$i]['table_uiprefs'] = 'pma__table_uiprefs';
$cfg['Servers'][$i]['users'] = 'pma__users';
$cfg['Servers'][$i]['usergroups'] = 'pma__usergroups';
$cfg['Servers'][$i]['navigationhiding'] = 'pma__navigationhiding';
$cfg['Servers'][$i]['savedsearches'] = 'pma__savedsearches';
$cfg['Servers'][$i]['central_columns'] = 'pma__central_columns';
$cfg['Servers'][$i]['designer_settings'] = 'pma__designer_settings';
$cfg['Servers'][$i]['export_templates'] = 'pma__export_templates';
$cfg['Servers'][$i]['favorite'] = 'pma__favorite';

/*
 * End of servers configuration
 */
?>`

**Comment by AshwinGinoria** [comment]
phpmyadmin installed successfully, Thank You very much for help!! :D
But now it automatically logs my into phpmyadmin Not prompting me for username/password (Is that expected Behavior ??
Also when I press logout it shows : 

![Screenshot from 2019-11-18 04-08-38](https://user-images.githubusercontent.com/43852150/69015421-a6c8d200-09b9-11ea-81f3-18ab6ef9106a.png)

**Comment by Tithugues** [comment]
@AshwinGinoria , there is a message you sent (and probably deleted) telling that in your configuration, you have: `$cfg['Servers'][$i]['auth_type'] = 'config';`
So yes, it's the expected behavior that you don't need to log in.
For the log out… I don't know, but I'm pretty sure this is not something important for now.

Can you reach other pages? Do some actions? If yes and yes, then you should now be able to work on this ticket.
Were you able to reproduce the encountered error?

**Comment by AshwinGinoria** [comment]
No, The error wasn't Reproduced my machine.
I followed the steps as given in the bug description and I think I got the edit view as expected (Not sure what edit view is ) .

![Screenshot from 2019-11-18 12-34-47](https://user-images.githubusercontent.com/43852150/69031422-1b266400-0a00-11ea-8d98-50a5727d48f9.png)

**Comment by Tithugues** [comment]
I think there is something the author didn't say: I guess he was using PHP7.3 (or upper).
Which one do you use?

**Comment by AshwinGinoria** [comment]
I am currently using PHP7.2

**Comment by AshwinGinoria** [comment]
Not able to reproduce even with php7.3.11 .

**Comment by Tithugues** [comment]
@williamdes , as 4.9 doesn't have strict mode, I think this should only be done from 5.0, don't you think?

**Comment by williamdes** [comment]
@Tithugues I would prefer to have the fix on 4.9 to avoid having to do a cherry-pick if it is needed some day

You can add comments on the PR if you want me to do a change a file when merging in 5.0

**Comment by Tithugues** [comment]
@AshwinGinoria , so I guess that you have to:
* do the changes on the 5.0
* apply them on 4.9
* check that it still works on 4.9
* submit your PR targetting the 4.9

**Comment by AshwinGinoria** [comment]
I am on branch QA_5_0 and my php version is 7.3.11. On following the steps to reproduce the error I am getting Internal Server Error.

![Screenshot from 2019-11-19 08-19-32](https://user-images.githubusercontent.com/43852150/69112163-9b52d500-0aa5-11ea-8a9f-1b5972726c03.png)

**Comment by Tithugues** [comment]
@AshwinGinoria , I would say you reproduced the error. You should have the PHP logs somewhere. Are the logs reporting the same error than the one described in this issue?

**Comment by AshwinGinoria** [comment]
@Tithugues the php log is showing the error mentioned above in the issue. I'll now get started on actually understanding the error. Thank You !! :D

**Comment by Tithugues** [comment]
@AshwinGinoria , any news regarding this error please? Do you expect to fix it or can someone else fix it?
Thank you!

**Comment by AshwinGinoria** [comment]
@Tithugues Sorry for replying so late !!
My end semester Exams Started Last week so I couldn't give much time to this. But I've spent some time understanding the relevant part of the code and I think I understand most of it. My Exams will be over This week, so can I work on it after that ??

**Comment by Tithugues** [comment]
@AshwinGinoria , thank you for the answer.
Unless the pMA team expect this bug to be fixed earlier than beginning of december, I'll wait for this time before trying to provide a fix.

**Comment by williamdes** [comment]
@ibennetch does it seem okay to you ?

**Comment by AshwinGinoria** [comment]
Hello, Sorry for being so late.
I've fixed the issue on 5.0, Now how should I proceed with applying it on 4.9 ??

**Comment by williamdes** [comment]
Hello @AshwinGinoria you can fix it on 4.9 branch and open a pull-request ;)
